src/main/java/com/passbook/sparkeighteen/service/TransactionService.java

```diff
@@ -0,0 +1,81 @@
+package com.passbook.sparkeighteen.service;
+
+import com.passbook.sparkeighteen.peristence.POJO.TransactionRequest;
+import com.passbook.sparkeighteen.peristence.POJO.TransactionResponse;
+import com.passbook.sparkeighteen.peristence.POJO.TransactionType;
+import com.passbook.sparkeighteen.peristence.entity.TransactionEntity;
+import com.passbook.sparkeighteen.peristence.entity.UserEntity;
+import com.passbook.sparkeighteen.peristence.repository.TransactionRepository;
+import com.passbook.sparkeighteen.peristence.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TransactionService {
+
+    private final TransactionRepository transactionRepository;
+    private final UserRepository userRepository;
+
+    public TransactionService(TransactionRepository transactionRepository, UserRepository userRepository) {
+        this.transactionRepository = transactionRepository;
+        this.userRepository = userRepository;
+    }
+
+    public TransactionResponse transact(Integer userID, TransactionRequest request) throws Exception {
+
+        Optional<UserEntity> optionalUser = userRepository.findById(userID);
+        if (optionalUser.isEmpty())
+            return TransactionResponse.builder()
+                    .message("User not found. Please use a registered user")
+                    .build();
+
+        UserEntity user = optionalUser.get();
+
+        Optional<TransactionEntity> optionalTxn = transaction(user, request);
+
+        if (optionalTxn.isEmpty())
+            return TransactionResponse.builder()
+                    .message("Deposit amount cannot be 0 or less.")
+                    .build();
+
+        TransactionEntity transaction = optionalTxn.get();
+
+        return TransactionResponse.builder()
+                .amount(transaction.getAmount())
+                .note(transaction.getNote())
+                .time(transaction.getTime())
+                .closingBalance(transaction.getClosingBalance())
+                .transactionType(transaction.getTransactionType())
+                .message(String.format("Your A/C XXXXX has a %s by Rs %.2f on %s ", transaction.getTransactionType(), transaction.getAmount(), transaction.getTime()))
+                .build();
+    }
+
+    private Optional<TransactionEntity> transaction(UserEntity user, TransactionRequest request) {
```

Comment:
what is this method doing?

Reply:
Updated, As perviously discussed need this to be closed